src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -326,20 +372,32 @@ const useCoreChatSystemHook = () => {
   React.useEffect(() => {
     if (!lastAnonymousMessage || !localDb) return;
 
-    const {type} = lastAnonymousMessage;
+    const {type, message} = lastAnonymousMessage;
     if (type === 'health.check') return;
+    if (processedMessageId === message?.id) return;
+
     if (type === 'notification.message_new') {
-      saveChannelListData(lastAnonymousMessage, ANONYMOUS).catch((e) => console.log(e));
+      // to prevent being processed twice (websocket response appears twice)
+      setProcessedMessageId(message?.id);
+      saveChannelListData(lastAnonymousMessage, ANONYMOUS)
+        .catch((e) => console.log(e))
+        .finally(() => setProcessedMessageId(null));
     }
   }, [JSON.stringify(lastAnonymousMessage), localDb]);
 
   React.useEffect(() => {
     if (!lastSignedMessage || !localDb) return;
 
-    const {type} = lastSignedMessage;
+    const {type, message} = lastSignedMessage;
     if (type === 'health.check') return;
+    if (processedMessageId === message?.id) return;
+
     if (type === 'notification.message_new' || type === 'notification.added_to_channel') {
-      saveChannelListData(lastSignedMessage, SIGNED).catch((e) => console.log(e));
+      // to prevent being processed twice (websocket response appears twice)
+      setProcessedMessageId(message?.id);
+      saveChannelListData(lastSignedMessage, SIGNED)
+        .catch((e) => console.log(e))
+        .finally(() => setProcessedMessageId(null));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a repetition in handling the lastAnonymousMessage and lastSignedMessage. You could create a helper function to handle these similar operations and reduce code duplication.
```diff
- React.useEffect(() => {
-     if (!lastAnonymousMessage || !localDb) return;
- 
-     const {type, message} = lastAnonymousMessage;
-      if (type === 'health.check') return;
-     if (processedMessageId === message?.id) return;
- 
-      if (type === 'notification.message_new') {
-       // to prevent being processed twice (websocket response appears twice)
-       setProcessedMessageId(message?.id);
-       saveChannelListData(lastAnonymousMessage, ANONYMOUS)
-         .catch((e) => console.log(e))
-         .finally(() => setProcessedMessageId(null));
-      }
-    }, [JSON.stringify(lastAnonymousMessage), localDb]);
-  
-    React.useEffect(() => {
-      if (!lastSignedMessage || !localDb) return;
-  
-     const {type, message} = lastSignedMessage;
-      if (type === 'health.check') return;
-     if (processedMessageId === message?.id) return;
- 
-      if (type === 'notification.message_new' || type === 'notification.added_to_channel') {
-       // to prevent being processed twice (websocket response appears twice)
-       setProcessedMessageId(message?.id);
-       saveChannelListData(lastSignedMessage, SIGNED)
-         .catch((e) => console.log(e))
-         .finally(() => setProcessedMessageId(null));
-     }
-   }, [JSON.stringify(lastSignedMessage), localDb]);

+ const handleNewMessage = (messageData: GetstreamWebsocket, channelType: MyChannelType) => {
+   const {type, message} = messageData;
+   if (type === 'health.check') return;
+   if (processedMessageId === message?.id) return;
+
+   if (type === 'notification.message_new' || type === 'notification.added_to_channel') {
+     // to prevent being processed twice (websocket response appears twice)
+     setProcessedMessageId(message?.id);
+     saveChannelListData(messageData, channelType)
+       .catch((e) => console.log(e))
+       .finally(() => setProcessedMessageId(null));
+   }
+ };
+
+ React.useEffect(() => {
+   if (!lastAnonymousMessage || !localDb) return;
+   handleNewMessage(lastAnonymousMessage, ANONYMOUS);
+ }, [JSON.stringify(lastAnonymousMessage), localDb]);
+ 
+ React.useEffect(() => {
+   if (!lastSignedMessage || !localDb) return;
+   handleNewMessage(lastSignedMessage, SIGNED);
+ }, [JSON.stringify(lastSignedMessage), localDb]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->